Dockerfile

```diff
@@ -19,10 +19,10 @@ COPY --from=builder ${FROM_DIRECTORY}/consoles/                             /usr
 
 RUN ln -s /usr/share/prometheus/console_libraries /usr/share/prometheus/consoles/ /etc/prometheus/
 RUN mkdir -p /prometheus && \
-    chown -R nobody:nobody etc/prometheus /prometheus
+    chgrp -R 0 /etc/prometheus /prometheus && \
+    chmod -R g=u /etc/prometheus /prometheus
 
 USER       nobody
 EXPOSE     9090
-VOLUME     [ "/prometheus" ]
```

Comment:
I'd like to keep the configuration in sync with upstream prometheus as much as possible, so I'll probably wait for https://github.com/prometheus/prometheus/pull/5050 to be merged first.

In general it seems that this case is a bit different than https://github.com/kubernetes/kube-state-metrics/pull/471, since the /prometheus directory contains data that should normally be persisted between container restarts.

Reply:
What this change will help is to avoid creating temporary volumes (i.e. `/var/lib/docker`) while user deliberately do not want to persist data and also helps with deploying this image on cloud providers who have restricted using VOLUME instruction / temporary volumes on host disk.